client-react/src/pages/app/deployment-center/code/DeploymentCenterCodeSourceAndBuild.tsx

```diff
@@ -219,9 +222,40 @@ const DeploymentCenterCodeSourceAndBuild: React.FC<DeploymentCenterFieldProps<De
     );
   };
 
+  const openConfigurationBlade = async () => {
+    const result = await portalContext.openBlade({
+      detailBlade: 'SiteConfigSettingsFrameBladeReact',
+      extension: 'WebsitesExtension',
+      detailBladeInputs: {
+        id: deploymentCenterContext.resourceId,
+      },
+    });
+
+    if (result) {
+      deploymentCenterContext.refresh();
+    }
+  };
+
+  useEffect(() => {
+    const isGitHubActionsOrKuduBuild =
+      selectedBuild === BuildProvider.GitHubAction || selectedBuild === BuildProvider.AppServiceBuildService;
+    setShowBasicAuthError(isGitHubActionsOrKuduBuild && !deploymentCenterPublishingContext.basicPublishingCredentialsPolicies?.scm.allow);
+  }, [selectedBuild, deploymentCenterPublishingContext.basicPublishingCredentialsPolicies?.scm.allow]);
```

Comment:
Use `useMemo` instead of `useState` and `useEffect`:

Add `useMemo` import to line 1's 'react' import.

```suggestion
  const showBasicAuthError = useMemo(() => {
    const isGitHubActionsOrKuduBuild =
      selectedBuild === BuildProvider.GitHubAction || selectedBuild === BuildProvider.AppServiceBuildService;
    return isGitHubActionsOrKuduBuild && !deploymentCenterPublishingContext.basicPublishingCredentialsPolicies?.scm.allow;
  }, [selectedBuild, deploymentCenterPublishingContext.basicPublishingCredentialsPolicies?.scm.allow]);
```